### PR TITLE
Fix issue with requests received before srv_conf section is loaded

### PR DESCRIPTION
**Pull request by SozinM**
Some requests (from my observations their size was always zero) come before srv_conf loads and it causes segfault.
I added not null checks to ensure we don't run into this issue again and moved return statement in case size==0 up so we don't execute extra code in this case.

### CODE REVIEW

**Review by cthtuf — DISMISSED** [review]

**Review by serjvanilla — DISMISSED** [review]

**Review by cthtuf — APPROVED** [review]

**Review by serjvanilla — APPROVED** [review]

File: docker/test_e2e.sh [hunk 1]
```diff
@@ -6,13 +6,12 @@ set -exo pipefail
...
-| tr -d \\n \
-| websocat --text -1 ws://${ENDPOINT}
+echo "asdasd" | websocat --text ws://${ENDPOINT}
```
Comment (Qjawko) [inline docker/test_e2e.sh:15]: Why did you change it?

Comment (SozinM) [reply docker/test_e2e.sh:15]: Because:
1. Behavior of the echo server I used (jmalloc/echo-server:0.3.3) is not compatible with -1 options from websocat as this server sends an additional message in the beginning
2. I just made the string smaller to keep it simple and short. There is no need for a full "lorem ipsum ..." thing.
3. I removed `tr -d \\n \` as we don't really need it
